src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -158,25 +180,18 @@ def __call__(
 
         return token, logits
 
-    def transfer_cache_state(self, engine: "NLDecoderEngine"):
+    def transfer_cache_storage(self, cache_storage: KVCacheSessionStorage):
         """
-        Transfers the kv cache state and the number of tokens processed
-        information from another NLDecoderEngine. Call this method when
-        you want to transfer the kv cache state from one engine to another.
+        Transfers the kv cache storage to the engine. Call this method when
+        you want to transfer the kv cache storage from e.g. one engine to another.
 
-        :param engine: The `NLDecoderEngine` to transfer the kv cache state
-            from
+        :param cache_storage: The cache storage to transfer the state from
         """
-        state = engine.kv_cache.cached_inputs
-
-        self.kv_cache.setup_session(
-            session_id=self._session_id,
-            state=state,
-            sequence_length=self.sequence_length,
-            freeze_first_position=self._freeze_first_position,
+        _LOGGER.debug(
+            f"Transferring cache storage {cache_storage} to the engine {self.engine}"
         )
-        # maybe set as a property for more control?
-        self._num_tokens = engine._num_tokens
+
+        self.kv_cache_storage = cache_storage
```

Comment:
how to we handle the fact that different engines will expect different cache shapes? i.e. we can't just set a pipeline with length 64 cache to have the same cache as a single token pipeline w/ a 127 length cache

Reply:
This is a valid remark, but let's push this requirement until we have the causal mask support. There is no way for us to properly test it for now.